lib/src/request_processor.cc

```diff
@@ -184,6 +185,8 @@ RequestProcessor::RequestProcessor(std::shared_ptr<PXPConnector> connector_ptr,
     loadInternalModules();
 
     if (!agent_configuration.modules_dir.empty()) {
+        // Tasks currently uses the modules_dir to avoid having to know pxp-agent's install location.
+        modules_["task"] = std::make_shared<Modules::Task>(agent_configuration.modules_dir, spool_dir_path_);
```

Comment:
This seems to break a bunch unit tests because it can't find the `task` binary. Moving metadata into the internal module would address this, but would not enable unit testing the task module.